spirv_cross.cpp

```diff
@@ -1094,6 +1094,22 @@ const SPIRType &Compiler::get_variable_data_type(const SPIRVariable &var) const
 	return get<SPIRType>(get_variable_data_type_id(var));
 }
 
+SPIRType &Compiler::get_variable_element_type(const SPIRVariable &var)
+{
+	SPIRType *type = &get_variable_data_type(var);
+	if (is_array(*type))
```

Comment:
The result will still be an array if the type is array-of-arrays. Is this intentional?

Reply:
From reading the code which uses it, it seems intentional, just want to confirm this in case there are some bizzarro cases for array-of-arrays ...

Reply:
Yeah, this is intentional. It's to strip off the first layer of array, since per-vertex I/O in a tessellation control shader is done to arrays.